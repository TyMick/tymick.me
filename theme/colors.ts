import { merge } from "lodash";
import chroma from "chroma-js";

const colors: any = {
  white: "#fff",
  gray100: "#f8f9fa",
  gray200: "#e9ecef",
  gray300: "#dee2e6",
  gray400: "#ced4da",
  gray500: "#adb5bd",
  gray600: "#6c757d",
  gray700: "#495057",
  gray800: "#343a40",
  gray900: "#212529",
  black: "#000",
  blue: "#007bff",
  indigo: "#6610f2",
  purple: "#6f42c1",
  pink: "#e83e8c",
  red: "#dc3545",
  orange: "#fd7e14",
  yellow: "#ffc107",
  green: "#28a745",
  teal: "#20c997",
  cyan: "#17a2b8",
};
merge(colors, {
  gray: colors.gray600,
  grayDark: colors.gray800,
  primary: colors.blue,
  secondary: colors.gray600,
  success: colors.green,
  info: colors.cyan,
  warning: colors.yellow,
  danger: colors.red,
  light: colors.gray100,
  dark: colors.gray800,
});
merge(colors, {
  light: {
    primary: chroma(colors.primary).brighten().hex(),
  },
});
merge(colors, {
  light: {
    primaryHoverFocus: chroma(colors.light.primary).brighten(0.5).hex(),
  },
});
export default colors;
