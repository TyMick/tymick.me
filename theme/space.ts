import type { SpaceScale } from "styled-components";

const space: SpaceScale = ["0", "0.25rem", "0.5rem", "1rem", "1.5rem", "3rem"];
const numNonNegativeElements = (space as string[]).length;
for (let i = 1; i < numNonNegativeElements; i++) {
  space[`-${i}`] = `-${space[i]}`;
}

export default space;
