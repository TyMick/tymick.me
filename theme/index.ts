import colors from "./colors";
import space from "./space";

const theme = { colors, space };
export default theme;
