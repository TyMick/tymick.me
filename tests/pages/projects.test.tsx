import React from "react";
import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";
import theme from "../../theme";
import Projects from "../../pages/projects";

describe("Projects", () => {
  it("matches snapshot", async () => {
    expect(
      renderer
        .create(
          <ThemeProvider theme={theme}>
            <Projects />
          </ThemeProvider>
        )
        .toJSON()
    ).toMatchSnapshot();
  });
});
