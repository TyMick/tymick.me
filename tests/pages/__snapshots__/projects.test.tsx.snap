// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects matches snapshot 1`] = `
.c0 {
  display: inline-block;
  max-width: 200px;
  width: 100%;
}

.c1 {
  margin-bottom: -0.25rem;
}

.c1 ul {
  margin-top: 0.25rem;
}

.c1 li {
  margin-bottom: 0.25rem;
}

<div
  className="cap-lg-49 container"
>
  <div
    className="align-items-center mb-3 mb-sm-5 row"
  >
    <div
      className="mr-md-4 mb-2 mb-sm-0 text-center text-sm-left col-sm-auto col-12"
    >
      <div
        className="c0 image-wrapper"
        width="100%"
      >
        <span
          style={
            {
              "background": "none",
              "border": 0,
              "boxSizing": "border-box",
              "display": "inline-block",
              "height": "initial",
              "margin": 0,
              "maxWidth": "100%",
              "opacity": 1,
              "overflow": "hidden",
              "padding": 0,
              "position": "relative",
              "width": "initial",
            }
          }
        >
          <span
            style={
              {
                "background": "none",
                "border": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": "initial",
                "margin": 0,
                "maxWidth": "100%",
                "opacity": 1,
                "padding": 0,
                "width": "initial",
              }
            }
          >
            <img
              alt=""
              aria-hidden={true}
              src="data:image/svg+xml,%3csvg%20xmlns=%27http://www.w3.org/2000/svg%27%20version=%271.1%27%20width=%2724%27%20height=%2724%27/%3e"
              style={
                {
                  "background": "none",
                  "border": 0,
                  "display": "block",
                  "height": "initial",
                  "margin": 0,
                  "maxWidth": "100%",
                  "opacity": 1,
                  "padding": 0,
                  "width": "initial",
                }
              }
            />
          </span>
          <img
            alt="Ty hiking in Joshua Tree National Park"
            className="rounded-circle"
            data-nimg="intrinsic"
            decoding="async"
            onError={[Function]}
            onLoad={[Function]}
            src="/_next/image?url=%2Fimg.jpg&w=48&q=75"
            srcSet="/_next/image?url=%2Fimg.jpg&w=32&q=75 1x, /_next/image?url=%2Fimg.jpg&w=48&q=75 2x"
            style={
              {
                "border": "none",
                "bottom": 0,
                "boxSizing": "border-box",
                "display": "block",
                "height": 0,
                "left": 0,
                "margin": "auto",
                "maxHeight": "100%",
                "maxWidth": "100%",
                "minHeight": "100%",
                "minWidth": "100%",
                "objectFit": undefined,
                "objectPosition": undefined,
                "padding": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
                "width": 0,
              }
            }
          />
        </span>
      </div>
    </div>
    <div
      className="text-center text-sm-left col-sm col-12"
    >
      <div
        className="greeting"
      >
        Hi! I’m Ty.
      </div>
    </div>
  </div>
  <h1
    className="h2"
  >
    A few things I’ve made:
  </h1>
  <ul
    className="c1"
  >
    <li>
      TyMick.me (
      <a
        href="https://github.com/TyMick/tymick.me"
      >
        source code
      </a>
      )
    </li>
    <li>
      <a
        href="https://rtsimulator.com"
      >
        <var
          dangerouslySetInnerHTML={
            {
              "__html": "",
            }
          }
        />
         Simulator
      </a>
       
      – An interactive data simulation that demonstrates why it’s so important to keep COVID-19’s reproduction number below 1.
    </li>
    <li>
      React hooks
      <ul
        className="c1"
      >
        <li>
          <a
            href="https://www.npmjs.com/package/use-window-width-breakpoints"
          >
            useWindowWidthBreakpoints
          </a>
        </li>
        <li>
          <a
            href="https://www.npmjs.com/package/use-window-orientation"
          >
            useWindowOrientation
          </a>
        </li>
      </ul>
    </li>
  </ul>
</div>
`;
